dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java

```diff
@@ -0,0 +1,313 @@
+package com.dotmarketing.portlets.browser;
+
+import static com.dotcms.util.CollectionsUtils.list;
+import static com.dotmarketing.util.StringUtils.builder;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.contenttype.exception.NotFoundInDbException;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.HostFolderField;
+import com.dotcms.rest.exception.NotFoundException;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.portlets.structure.model.Field.FieldType;
+import com.dotmarketing.portlets.structure.model.FieldVariable;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class BrowserUtil {
```

Comment:
Add some doc. What is the idea of this Util? Add Javadoc to methods.

Reply:
done https://github.com/dotCMS/core/pull/21224/commits/4d0aadeff34b620c13f5c90ccf378cffc2960c7a